tests/functional/b/bad_dunder_name.py

```diff
@@ -0,0 +1,50 @@
+# pylint: disable=missing-module-docstring, missing-class-docstring,
+# pylint: disable=missing-function-docstring, unused-private-member
+
+
+class Apples:
+    __slots__ = ("a", "b")
+
+    def __hello__(self):  # [bad-dunder-name]
+        # not one of the explicitly defined dunder name methods
+        print("hello")
+
+    def hello(self):
+        print("hello")
+
+    def __init__(self):
+        pass
+
+    def init(self):
+        # valid name even though someone could accidentally mean __init__
+        pass
+
+    def __init_(self):  # [bad-dunder-name]
+        # author likely unintentionally misspelled the correct init dunder.
+        pass
+
+    def _init_(self):
+        # protected, not private
+        pass
+
+    def ___neg__(self):  # [bad-dunder-name]
+        # author likely accidentally added an additional `_`
+        pass
+
+    def __inv__(self):  # [bad-dunder-name]
+        # author likely meant to call the invert dunder method
+        pass
+
```

Comment:
```suggestion


    def __user_defined_good_dunder_name__(self):
        print("Whatever the people want they get.")
```

Will need a rc file with:
```ini
good_dunder_name = __user_defined_good_dunder_name__, __something_else__
```

Reply:
see comment above. If you don't think it's premature, I"ll add it.